client/state/data-layer/wpcom-http/utils.js

```diff
@@ -220,7 +220,7 @@ export const dispatchRequestEx = options => {
 		// create the low-level action we want to dispatch
 		const requestAction = createRequestAction( options, action );
 		// dispatch the low level action (if any was created) and return the result
-		return requestAction ? store.dispatch( requestAction ) : undefined;
+		return requestAction ? [].concat( requestAction ).forEach( store.dispatch ) : undefined;
```

Comment:
should this be `.map` instead of `.forEach` so that the return value can be meaningful?

I have a feeling that @jsnajdr would want this to eventually return `Array<Promise>`

Reply:
Yes please 🙂 Also, when there is only one action returned, not an array, return `rv` instead of `[ rv ]`.

Also, I was thinking: what if we allowed `dispatchRequestEx` to return not only array, but anything iterable? Like React accepts anything that has a `Symbol.iterator` as `children`:
```jsx
const generateChildren = function*() {
  yield 'header';
  yield <Body />;
  yield 'footer';
};
return <div>{ generateChildren() }</div>;
```
But that's just a crazy idea, completely optional. I don't have any use for it right now :wink:

Reply:
like it, will leave it for the future :wink:

Reply:
preserved and returned in 78b84867e